src/back/server.py

```diff
@@ -6,171 +6,133 @@
         return web.Response(
             content_type="application/json",
-            headers={"Access-Control-Allow-Origin": "*", 
-                        "Access-Control-Allow-Credentials": "true", 
-                        "Access-Control-Allow-Methods": "POST, GET, OPTIONS", 
-                        "Access-Control-Allow-Headers": "Content-Type"},
+            headers={
+                "Access-Control-Allow-Origin": "*",
+                "Access-Control-Allow-Credentials": "true",
+                "Access-Control-Allow-Methods": "POST, GET, OPTIONS",
+                "Access-Control-Allow-Headers": "Content-Type"
+            },
         )
 
-    params = await request.json()
-    description = RTCSessionDescription(sdp=params["sdp"], type=params["type"])
+    body = await request.json()
+    try:
+        validate(instance=body, schema=consumer_schema)
+    except ValidationError as error:
+        raise web.HTTPBadRequest(reason=error.message)
 
+    description = RTCSessionDescription(sdp=body["sdp"], type=body["type"])
     pc = RTCPeerConnection()
     pc_id = "PeerConnection(%s)" % uuid.uuid4()
-    pcs.add(pc)
+    peer_connections.add(pc)
 
     def log_info(msg, *args):
         logger.info(pc_id + " " + msg, *args)
 
-    log_info("Track %s sent", consumer_track.kind)
-    pc.addTrack(
-        # VideoTransformTrack(relay.subscribe(consumer_track), transform=params["video_transform"])
-        VideoTransformTrack(relay.subscribe(consumer_track), transform=annotation)
-    )
+    video_transform = NO_TRANSFORM if not body["video_transform"] else body["video_transform"].lower()
+    try:
+        transformed_track = apply_transform(source_video, transform=video_transform)
+    except UnsupportedTransform as error:
+        raise web.HTTPBadRequest(reason=error.message)
+
+    pc.addTrack(relay.subscribe(transformed_track))
+    log_info("Track %s sent", source_video.kind)
 
     await pc.setRemoteDescription(description)
     answer = await pc.createAnswer()
     await pc.setLocalDescription(answer)
 
     return web.Response(
         content_type="application/json",
-        text=json.dumps(
-            {"sdp": pc.localDescription.sdp, "type": pc.localDescription.type}
-        ),
-        headers={"Access-Control-Allow-Origin": "*", 
-            "Access-Control-Allow-Credentials": "true", 
-            "Access-Control-Allow-Methods": "POST, GET, OPTIONS", 
-            "Access-Control-Allow-Headers": "Content-Type"},
+        text=json.dumps({
+            "sdp": pc.localDescription.sdp,
+            "type": pc.localDescription.type
+        }),
+        headers={
+            "Access-Control-Allow-Origin": "*",
+            "Access-Control-Allow-Credentials": "true",
+            "Access-Control-Allow-Methods": "POST, GET, OPTIONS",
+            "Access-Control-Allow-Headers": "Content-Type"
+        },
     )
 
 
 async def broadcast(request):
     if request.method == "OPTIONS":
         return web.Response(
```

Comment:
both broadcaster and consumer web response are in the same format, put them into  a function to reduce code duplication.

Reply:
Wont fix: after introducing a constant for the header, we're just saving two lines here inside a return statement